### PR TITLE
Holographic Janitorial Signs!

### PR DESCRIPTION
**Pull request by Iamgoofball**
itshappeningthesigns
exactly what the title says, the sign creator can destroy any signs it makes by hitting the sign with the creator.

### CODE REVIEW

**Comment by Atskadan** [comment]
:+1:

File: code/defines/obj/weapon.dm [hunk 1]
```diff
@@ -210,23 +210,6 @@
...
-	throw_speed = 2
-	throw_range = 5
-	w_class = 2.0
-	attack_verb = list("warned", "cautioned", "smashed")
-
-/obj/item/weapon/caution/cone
-	desc = "This cone is trying to warn you of something!"
-	name = "warning cone"
-	icon_state = "cone"
-
```
Comment (Atskadan) [inline code/defines/obj/weapon.dm:229]: woah their freind.

no need to get rid of the old ones

Comment (RemieRichards) [inline code/defines/obj/weapon.dm:229]: He didn't, See new file.

Comment (Atskadan) [inline code/defines/obj/weapon.dm:229]: stop bullying me

File: code/modules/janitor/holosign.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	desc = "The words flicker as if they mean nothing."
+	name = "wet floor sign"
+	icon_state = "holosign"
+	anchored = 1
+
+/obj/item/weapon/janitor/caution/holograph/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/weapon/janitor/holosign_creator))
+		user << "You use [I] to destroy [src]."
+		del(src)
```
Comment (Petethegoat) [inline code/modules/janitor/holosign.dm]: qdel(src)

File: code/modules/janitor/holosign.dm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+/obj/item/weapon/janitor/caution/holograph
+	desc = "The words flicker as if they mean nothing."
+	name = "wet floor sign"
+	icon_state = "holosign"
+	anchored = 1
+
+/obj/item/weapon/janitor/caution/holograph/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/weapon/janitor/holosign_creator))
+		user << "You use [I] to destroy [src]."
```
Comment (Petethegoat) [inline code/modules/janitor/holosign.dm]: user << "<span class='notice'>You use [I] to destroy [src].</span>"

File: code/modules/janitor/holosign.dm [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	flags = CONDUCT
+	force = 5.0
+	w_class = 2.0
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	origin_tech = "programming=3"
+/obj/item/weapon/janitor/holosign_creator/attack_self(mob/user as mob)
+	..()
+	user << "You create a holographic sign with the device."
```
Comment (Petethegoat) [inline code/modules/janitor/holosign.dm]: user << "<span class='notice'>You create a holographic sign with the device.</span>"

File: code/modules/janitor/holosign.dm [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/obj/item/weapon/janitor/holosign_creator
+	name = "\improper Holographic Sign Creator"
```
Comment (Petethegoat) [inline code/modules/janitor/holosign.dm]: just call it "holographic sign creator", no need for capitals at all.

File: code/modules/janitor/holosign.dm [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	force = 5.0
+	w_class = 2.0
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	origin_tech = "programming=3"
+/obj/item/weapon/janitor/holosign_creator/attack_self(mob/user as mob)
+	..()
+	user << "You create a holographic sign with the device."
+	new /obj/item/weapon/janitor/caution/holograph(get_turf(user))
```
Comment (Petethegoat) [inline code/modules/janitor/holosign.dm]: might be better to have it pool the signs and project a limited number at once? say ten signs max, with older signs disappearing when you add more than 10.

**Comment by Petethegoat** [comment]
Also not sure if the /janitor subtype is worth it, I think we're generally trying to stay away from purely organisational paths. Plus it creates some extra incompatibility with /tg/ for maps and stuff.

I'll make a PR to your branch with the changes I've outlined, I think. Looking forward to seeing this stuff ingame though, janitor can never get enough cool gizmos.

**Comment by Petethegoat** [comment]
Made a PR here: https://github.com/Iamgoofball/NTstation13/pull/1

Decided not to bother with pooling in the end, it's probably not worth the effort.

**Pull request merged**